Pre5 CHANGELOG and version bump

jasonrclark opened the pull request:
Bit ahead of things, but prepped a CHANGELOG since we missed the bit on a few of the merges along the way.

Still waiting on #1170  and #1171, both of which look to be close to :ship: 

Once those are in, will do a final big run-through of current samples. Might look at busted samples too and see whether any of those have magically gotten better since last time we looked...

New :athletic_shoe:, cool like :candy:

PragTob commented on CHANGELOG:
```diff
@@ -1,3 +1,34 @@
+* Fix installation for jruby-complete [fa65ab0]
+* Remove the cached app template from repo (and gem) [68bd6a8]
+* Fix handling of explicitly set text widths [8ca2096]
+* Don't consider backgrounds and borders in positioning [18d337b]
```
wasn't there another rather mean bug that actually stopped the backend from being initialized? So A gem install was sort of broken?

jasonrclark commented:
Don't see a bug that was around between gem releases like that. If we broke and fixed it since pre4 I'm not inclined to make a note of it necessarily.

That said, I noticed https://github.com/shoes/shoes4/pull/1147 which definitely deserves calling out. Maybe that's the thing you were remembering @PragTob ? That PR did have some bumps along the way if I remember correctly.

Also added the CoC. Worth a shout out you think? https://github.com/shoes/shoes4/pull/1144

Other changes of note https://github.com/shoes/shoes4/pull/1122 and https://github.com/shoes/shoes4/pull/1123

PragTob commented:
Yes #1147 / #1146 was the one I was worried about, especially since I reported it again in #1169 :P 

CoC definitely worth a shout. #1123 should also go in imo

jasonrclark commented:
On further inspection #1122 and #1123 were both already there in pre4. D'oh. Thinking might be worth marking PR's with their destination milestone in the future to make that more obvious at a glance when things merged.

Incoming updates for CoC (#1144) and running with just `require 'shoes'` (#1146)...

jasonrclark commented:
Anyone else want to do any testing before we put pre5 in the :gift: ? @PragTob @KCErb @AStrangeEnigma ?

AStrangeEnigma commented:
I'm good! Been using always on top for a personal project and it works like a charm :)

If you could, could you change AStrangeEnigma in the changelog to my full name? Richard McCormack (just added it to my github profile)

thanks! Cool to be part of a release like this :)

jasonrclark commented:
Glad to make the update Richard, and glad to have you using Shoes! a0749ad2ddaaa1b054f4694b2faef3af107aa8bc

KCErb commented:
I *want* to; we'll be driving back from Canada today so that might mean I can run things on the laptop. But it might also mean I won't get a chance until Monday. If you want to ship it before then that'd be fine. Great work Jason!

PragTob commented:
I'd want to fix #1174 before we land pre5 #noregressions

jasonrclark commented:
FWIW, looks like this was actually broken already in pre4, but I think I've got a fix in the works so will hold for it.

jasonrclark commented:
Bumped CHANGELOG for #1174 fixed via #1177

jasonrclark commented:
Hey all! Unless I hear otherwise, how's about we ship `pre5` this weekend? Will give till late in the day Sunday 12/6 (PST) for anyone to test and put a halt to the release :bullettrain_side:

PragTob commented:
:+1: sounds great, just wanted to check in with you when you wanted to do that :D

jasonrclark commented:
So my weekend didn't yield as much time as I expected, and with the recent influx of other issues, I'm gonna hold on this a little longer and see what we can do about those problems.

If any of them are 1) holdovers from `pre4` and 2) hard, then I'm in favor of shipping `pre5` as it stands. Any low-hanging fruit, though, may as well pick off.

Any objections?

PragTob commented:
#1180 is the only potentially low hanging fruit that comes to my mind (open to lots of other though!), everything else seems like rather a bunch of work. Also we should cut a release - if fix something critical, nothing bad about doing a quick release with just 2 or 3 fixes.

If you can't get to it - I'd have time this evening or tomorrow evening :)

jasonrclark commented:
Yeah, after poking around on the latest issue reports/etc. I'm thinking let's merge #1186 and #1189 and get this thing out the door!

#1180 I'd like to confirm the existing values with folks, and don't want to hold the train up on them.

jasonrclark commented:
CHANGELOG updated. @PragTob if you have time/desire, feel free to do any final check and release. Otherwise, I might be able to do it after company leaves tonight.

jasonrclark commented:
Oh silly Jason, thinking he'll have computer time after visitors. Hope for final pre5 test Monday night...

PragTob commented:
@jasonrclark I hope to beat you to it so you can some actual work done :D Yay running through all the samples... :D

PragTob commented:
So I did find a couple of issues browsing through all the samples... i.e. #1192 #1193 and #1194 .

I feel like I want to tackle/at least have a look at why #1193 and #1194 happen as they seem to be regressions... I'd set a latest release limit for next Sunday though.

jasonrclark commented:
I'm good with calling it a :ship: next weekend, whatever's made it in. Thanks for your time getting back to it! @PragTob

PragTob commented:
So from my side this is good to go with my 2 PRs in. I'm rather confident about them, but due to all the changes they probably warrant another full check through the samples.

jasonrclark commented:
:ship: :shipit: :rocket:

PragTob commented:
:star: :star2: :balloon: :tada:

KCErb commented:
Wow, just at the stroke of midnight, great job everyone!!!
